java/google/registry/rde/imports/XjcToDomainResourceConverter.java

```diff
@@ -293,16 +238,34 @@ private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
   }
 
   /** Converts {@link XjcRdeDomainTransferDataType} to {@link TransferData}. */
-  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data) {
+  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data,
+      DateTime domainExpiration) {
     if (data == null) {
       return TransferData.EMPTY;
     }
+    // If the transfer is pending, calculate the number of years to add to the domain expiration
+    // on approval of the transfer.
+    TransferStatus transferStatus = TRANSFER_STATUS_MAPPER.xmlToEnum(data.getTrStatus().value());
+    // Get new expiration date
+    DateTime newExpirationTime = domainExpiration;
+    if (transferStatus == TransferStatus.PENDING) {
+      // Default to domain expiration time plus one year if no expiration is specified
+      if (data.getExDate() == null) {
+        newExpirationTime = newExpirationTime.plusYears(1);
+      } else {
+        newExpirationTime = data.getExDate();
+      }
+    }
     return new TransferData.Builder()
-        .setTransferStatus(TRANSFER_STATUS_MAPPER.xmlToEnum(data.getTrStatus().value()))
+        .setTransferStatus(transferStatus)
         .setGainingClientId(data.getReRr().getValue())
         .setLosingClientId(data.getAcRr().getValue())
         .setTransferRequestTime(data.getReDate())
         .setPendingTransferExpirationTime(data.getAcDate())
+        // This will be wrong for domains that are not in pending transfer,
```

Comment:
interesting. I agree that it would be simpler to make this implicitly 1 year and eliminate the need for the field.